pkg/storageops/aws/aws.go

```diff
@@ -377,30 +360,47 @@ func (s *ec2Ops) available(v *ec2.Volume) bool {
 	return *v.State == ec2.VolumeStateAvailable
 }
 
-func (s *ec2Ops) Inspect(volumeIds []*string) ([]*ec2.Volume, error) {
+func (s *ec2Ops) GetDeviceID(vol interface{}) string {
+	v := vol.(*ec2.Volume)
+	return *v.VolumeId
+}
+
+func (s *ec2Ops) Inspect(volumeIds []*string) ([]interface{}, error) {
 	req := &ec2.DescribeVolumesInput{VolumeIds: volumeIds}
-	awsVols, err := s.ec2.DescribeVolumes(req)
+	resp, err := s.ec2.DescribeVolumes(req)
 	if err != nil {
 		return nil, err
 	}
-	return awsVols.Volumes, nil
+
+	var awsVols = make([]interface{}, len(resp.Volumes))
+	for i, v := range resp.Volumes {
+		awsVols[i] = v
+	}
+
+	return awsVols, nil
 }
 
-func (s *ec2Ops) Tags(v *ec2.Volume) map[string]string {
+func (s *ec2Ops) Tags(v interface{}) (map[string]string, error) {
```

Comment:
Can we just change the API itself to Tags(volumeID string)  instead of doing a refreshVol on an ec2Volume ?

Reply:
Changed API in incremental